README.md

```diff
@@ -4,569 +4,1111 @@
+- An utility method is added to allow for validations against the filename extension with `validate_extension :image, as: %w[jpg jpeg png]`
 
-```
-class Post < ActiveRecord::Base
-  include Saviour::Model
-  attach_file :image, PostImageUploader
-  attach_validation :image, :check_size
 
-  private
+### How to manage file removal from forms
+
+This feature can be implemented with a temporal flag in the model, which is exposed in the forms and passed via
+controllers, and a `before_update` to read the value and delete the attachment if present. For example, the
+`FileAttachment` module exposed in the previous point could be extended as such:
 
-  def check_size(contents, filename)
-    errors.add(:image, "must be smaller than 10Mb") if contents.bytesize >= 10.megabytes
+```ruby
+module FileAttachment
+  # ...
+  class_methods do
+    def attach_file_with_defaults(*args, &block)
+      attached_as = args[0]
+      # ...
+    
+      define_method("remove_#{attached_as}") do
+        instance_variable_get("@remove_#{attached_as}")
+      end
+
+      alias_method "remove_#{attached_as}?", "remove_#{attached_as}"
+
+      define_method("remove_#{attached_as}=") do |value|
+        instance_variable_set "@remove_#{attached_as}", ActiveRecord::Type::Boolean.new.cast(value)
+      end
+      
+      before_update do
+        send("remove_#{attached_as}!") if send("remove_#{attached_as}?")
+      end
+    end
   end
 end
 ```
 
-To improve reusability, validation blocks or methods will also receive a third argument (only if declared in your
-implementation). This third argument is a hash containing `attached_as` and `version` of the validating file.
+Then it can be used as:
 
+```ruby
+# This would be a controller code
+a = Post.find(42)
 
-## Active Record Lifecycle integration
+# Params received from a form
+a.update_attributes(remove_image: "t")
+```
 
-On `after_save` Saviour will upload the changed files attached to the current model, executing the processors as needed.
 
-On `after_destroy` Saviour will delete all the attached files and versions.
+### How to extract metadata from files
 
-On `validate` Saviour will execute the validations defined.
+You can use processors to accomplish this. Just be aware that processors run concurrently, so if you want to 
+persist you extracted information in the database probably you'll want to use `stash`, see [the section
+about stash feature for examples](#stash).
 
-When validations are defined, the assigned source will be readed only once. On validation time, it will be readed, passed
-to the validation blocks and cached. If the model is valid, the upload will happen from those cached contents. If there
-are no validations, the source will be readed only on upload time, after validating the model.
 
+### How to process files in background / delayed
 
-## FAQ
+As a previous warning note, pushing logics to be ran in the background, when they have visible consequences for the application, may
```

Comment:
logics -> logic
ran -> run

and too many `consequences`, any way to rephrase?